### DISCUSSION TITLE
Unable to launch application in QEMU on Windows

### DISCUSSION THREAD

**Discussion by hafizmqasim** [zephyrproject-rtos/zephyr#67690] [Q&A]
Hi 

I have successfully build the hello world application for QEMU Cortex A53 using the command
**west build -p always -b qemu_cortex_a53 samples\hello_world**

But when executed the following command
**west build -t run**

Following error is observed 

-- west build: running target run
[0/1] To exit from QEMU enter: 'CTRL+a, x'[QEMU] CPU: cortex-a53'QEMU-NOTFOUND' is not recognized as an internal or external command,
operable program or batch file.

FAILED: zephyr/CMakeFiles/run_qemu C:/Users/m3n1iz/zephyrproject/zephyr/build/zephyr/CMakeFiles/run_qemu
cmd.exe /C "cd /D C:\Users\m3n1iz\zephyrproject\zephyr\build && QEMU-NOTFOUND -cpu cortex-a53 -nographic -machine virt,secure=on,gic-version=3 -net none -pidfile qemu.pid -chardev stdio,id=con,mux=on -serial chardev:con -mon chardev=con,mode=readline -icount shift=4,align=off,sleep=on -rtc clock=vm -kernel C:/Users/m3n1iz/zephyrproject/zephyr/build/zephyr/zephyr.elf"
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: 'C:\Program Files\CMake\bin\cmake.EXE' --build 'C:\Users\m3n1iz\zephyrproject\zephyr\build' --target run

Can someone help? Is QEMU supported on Windows 11?

**Comment by hafizmqasim** [comment]
As per this page, QEMU is not supported on Windows?
https://docs.zephyrproject.org/latest/develop/application/index.html#running-in-an-emulator

Is there a way to run qemu_cortex_a53 zephyr.elf on Windows

I can see zephyr.[dts/elf/map] in "zephyrproject\zephyr\build\zephyr" but hello world dtb is not present? Is this file created during build?

**Reply by leenaarora18** [reply]
I'm also getting the same problem. Please help. Do we have to change the git hub repository of zephyrproject/zephyr/samples/hello 
world/prj.config file

**Reply by alexpaschoaletto** [reply]
> I'm also getting the same problem. Please help. Do we have to change the git hub repository of zephyrproject/zephyr/samples/hello world/prj.config file

Have you checked my tip in the comment below? It worked for me and mr Tace03, maybe will do for you.

**Reply by leenaarora18** [reply]
I saw that as of my understanding, we are clone the repositry from zephyrproject so changes mention **CONFIG_STDOUT_CONSOLE=y in prj.conf,**  where im supposed to do these changes.

**Reply by Tace03** [reply]
Your comment actually pointed out the correct place for changes though, you can add CONFIG_STDOUT_CONSOLE=y into the prj.conf residing at zephyrproject/zephyr/samples/hello world/ directory.

Works fine on Windows 10 for me previously.

**Comment by alexpaschoaletto** [comment]
I don't know if you found the answer by yourself by now, but I just ran through the same problem trying to execute the "hello world" example. As a sanity check I decided to test the default "syncronization" example [described in the docs](https://docs.zephyrproject.org/latest/boards/qemu/riscv32/doc/index.html) for all QEMU ports and found the missing piece in the puzzle seems to be `CONFIG_STDOUT_CONSOLE=y` in prj.conf, which is missing in "hello world". Adding this option solved it for me.

By the way, you also might experience `k_sleep()` not working as expected, which has already been described in a previous [issue](https://github.com/zephyrproject-rtos/zephyr/issues/29518). Setting `CONFIG_QEMU_ICOUNT=n` was the solution in my case.

**Reply by Tace03** [reply]
Hi, could you help me point out where the file prj.conf would stay in the directory? There seems to be many of them and I am not sure which one to edit

**Reply by alexpaschoaletto** [reply]
I can only speak from the perspective of a standalone project (which are the ones I work with). For those, the `prj.conf` should be a file located in the root of the project folder alongside an `app.overlay` and a `CMakeLists.txt` file. Your source code (`main.c` and application libraries) should be within the src folder. Basically something like this:

![image](https://github.com/zephyrproject-rtos/zephyr/assets/111513888/b5749396-d6ae-474a-8f96-b0e8f142760f)

So while the the `prj.conf` is meant to include/remove kernel features, the `app.overlay` is meant to specify devicetree changes to default settings (such as changing UART pins) and/or creating new, undefined cofigurations (e.g. assigning GPIOs).

**Reply by Tace03** [reply]
Hi, I added the config entry as you mentioned and it worked just fine.

I must say this is quite surprising from the side of zephyr as I thought samples should be fool-proof to get things started, but yeah will report this issue to the devs I suppose.
